### PR TITLE
Misc work

### PR DESCRIPTION
**Pull request by est77**
- Removed velvet brdf, it was unused. The sheen brdf is a better alternative.
- Added simple wrapper for OSL's shader compiler.
- Removed left over python 3 compatibility code.
- Renamed logging.py to avoid conflicts with Python's logging module.

### CODE REVIEW

File: src/appleseed/CMakeLists.txt [hunk 1]
```diff
@@ -1140,6 +1140,8 @@ set (renderer_kernel_shading_sources
     renderer/kernel/shading/oslshadergroupexec.h
     renderer/kernel/shading/oslshadingsystem.cpp
     renderer/kernel/shading/oslshadingsystem.h
+    renderer/kernel/shading/oslshadercompiler.cpp
```
Comment (dictoon) [inline src/appleseed/CMakeLists.txt]: Order alphabetically.

File: src/appleseed/renderer/kernel/shading/oslshadercompiler.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#define APPLESEED_RENDERER_KERNEL_SHADING_OSLSHADERCOMPILER_H
+
+// appleseed.foundation headers.
+#include "foundation/core/concepts/noncopyable.h"
+#include "foundation/utility/api/apistring.h"
+#include "foundation/utility/autoreleaseptr.h"
+
+// Standard headers.
+#include <cstddef>
+
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/shading/oslshadercompiler.h]: Remove one blank line.

File: src/appleseed/renderer/kernel/shading/oslshadercompiler.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    struct Impl;
+    Impl* impl;
+
+    explicit ShaderCompiler(const char* stdosl_path);
+    ~ShaderCompiler();
+};
+
+
+//
+// ShaderQuery factory.
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/shading/oslshadercompiler.h]: Fix comment.

File: src/appleseed/renderer/kernel/shading/oslshadercompiler.h [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef APPLESEED_RENDERER_KERNEL_SHADING_OSLSHADERCOMPILER_H
+#define APPLESEED_RENDERER_KERNEL_SHADING_OSLSHADERCOMPILER_H
+
+// appleseed.foundation headers.
+#include "foundation/core/concepts/noncopyable.h"
+#include "foundation/utility/api/apistring.h"
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/shading/oslshadercompiler.h]: I think we could live with a forward decl of `foundation::APIString` no?

**Comment by est77** [comment]
Fixed and rebased

**Pull request merged**